viewpump/src/main/java/io/github/inflationx/viewpump/ViewPumpLayoutInflater.java

```diff
@@ -0,0 +1,436 @@
+        setUpLayoutFactories(false);
+    }
+
+    protected ViewPumpLayoutInflater(LayoutInflater original, Context newContext, final boolean cloned) {
+        super(original, newContext);
+        nameAndAttrsViewCreator = new NameAndAttrsViewCreator(this);
+        parentAndNameAndAttrsViewCreator = new ParentAndNameAndAttrsViewCreator(this);
+        setUpLayoutFactories(cloned);
+    }
+
+    @Override
+    public LayoutInflater cloneInContext(Context newContext) {
+        return new ViewPumpLayoutInflater(this, newContext, true);
+    }
+
+    // ===
+    // Wrapping goodies
+    // ===
+
+
+    @Override
+    public View inflate(XmlPullParser parser, ViewGroup root, boolean attachToRoot) {
+        setPrivateFactoryInternal();
+        return super.inflate(parser, root, attachToRoot);
+    }
+
+    /**
+     * We don't want to unnecessary create/set our factories if there are none there. We try to be
+     * as lazy as possible.
+     */
+    private void setUpLayoutFactories(boolean cloned) {
+        if (cloned) return;
+        // If we are HC+ we get and set Factory2 otherwise we just wrap Factory1
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.HONEYCOMB) {
+            if (getFactory2() != null && !(getFactory2() instanceof WrapperFactory2)) {
+                // Sets both Factory/Factory2
+                setFactory2(getFactory2());
+            }
+        }
+        // We can do this as setFactory2 is used for both methods.
+        if (getFactory() != null && !(getFactory() instanceof WrapperFactory)) {
+            setFactory(getFactory());
+        }
+    }
+
+    @Override
+    public void setFactory(Factory factory) {
+        // Only set our factory and wrap calls to the Factory trying to be set!
+        if (!(factory instanceof WrapperFactory)) {
+            super.setFactory(new WrapperFactory(factory, this));
+        } else {
+            super.setFactory(factory);
+        }
+    }
+
+    @Override
+    @TargetApi(Build.VERSION_CODES.HONEYCOMB)
+    public void setFactory2(Factory2 factory2) {
+        // Only set our factory and wrap calls to the Factory2 trying to be set!
+        if (!(factory2 instanceof WrapperFactory2)) {
+//            LayoutInflaterCompat.setFactory(this, new WrapperFactory2(factory2, mViewPumpFactory));
+            super.setFactory2(new WrapperFactory2(factory2));
+        } else {
+            super.setFactory2(factory2);
+        }
+    }
+
+    private void setPrivateFactoryInternal() {
+        // Already tried to set the factory.
+        if (mSetPrivateFactory) return;
+        // Reflection (Or Old Device) skip.
+        if (!ViewPump.get().isReflection()) return;
+        // Skip if not attached to an activity.
+        if (!(getContext() instanceof Factory2)) {
+            mSetPrivateFactory = true;
+            return;
+        }
+
+        final Method setPrivateFactoryMethod = ReflectionUtils.getMethod(LayoutInflater.class, "setPrivateFactory");
```

Comment:
will add a TODO comment to cache the `Method`. please correct me if i've misunderstood